### ISSUE TITLE
Unexpected Segmentation fault

### ISSUE THREAD

**Issue by dapabe** [oven-sh/bun#13308]
### How can we reproduce the crash?

This is the main code causing the issue, honestly this quite unexpected, this only toggles boolean values inside an object, outside the main command line app the updated value is saved into a encrypted .json file

In case you need more context, this is the source code: 
https://github.com/dapabe/CLI-Pago_mis_servicios/tree/bun
run commands
```
bun build:dev
bun start:dev --skip-server
```

```ts
import { cancel, isCancel, select } from "@clack/prompts";
import picocolors from "picocolors";

export async function flagsPrompt(userData: IUserData, initialValue: Extract<IPromptAction, "exit"> | IFlagConfigOption = "exit") {
  const answer = await select<any, Extract<IPromptAction, "exit"> | IFlagConfigOption>({
    message: "Configuración",
    initialValue,
    options: [
      {
        label: "Volver",
        value: "exit",
      },
      {
        label: `Modo seguro: ${userData.flags.secure ? picocolors.green("Activado") : `${picocolors.red("Desactivado")} - ${picocolors.yellow("[Ten cuidado a quien muestras tus datos]")}`}`,
        value: "secure",
        hint: "Oculta o no el valor de los campos con información sensible",
      },
      {
        label: `Navegador activado: ${!userData.flags.headless ? picocolors.green("Activado") : `${picocolors.red("Desactivado")}`}`,
        value: "headless",
        hint: `Abre el navegador interno y muestra los pasos que se hace - ${picocolors.yellow("[Es recomendable que no se toque nada]")}`,
      },
    ],
  });

  if (isCancel(answer)) {
    cancel(SafeExitMessage);
    process.exit(0);
  }
  if (answer === "exit") return;

  userData.flags[answer] = !userData.flags[answer]

  await flagsPrompt(userData, answer)
}

```
![image](https://github.com/user-attachments/assets/fce3bd58-10d9-49e1-bebd-cedf3aa8eb56)

### Relevant log output

```shell
Bun Canary v1.1.22-canary.96 (df33f2b2) Windows x64
CPU: sse42 avx avx2
Args: "C:\Program Files\nodejs\\node_modules\bun\bin\bun.exe" "--env-file=.env.development" "--sourcemap" "dist/index.js"
Features: jsc Bun.stdin(2) Bun.stdout dotenv tsconfig_paths(2) tsconfig(2)
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:constants" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:module" "node:net" "node:os" "node:path" "node:process" "node:readline" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "utf-8-validate" "node:http2" "node:dgram"
Elapsed: 74862ms | User: 593ms | Sys: 140ms
RSS: 0.20GB | Peak: 0.22GB | Commit: 0.27GB | Faults: 78478

panic(main thread): Segmentation fault at address 0x5
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.22/wa2df33f2bA6ogggD___8uqm0B__2z4nxB___A2AK
```

### Stack Trace (bun.report)

Bun v1.1.22 ([`df33f2b`](<https://github.com/oven-sh/bun/tree/df33f2b2a264912c7a742d0c03587082415fbba6>)) on windows x86_64 [AutoCommand]

Segmentation fault at address 0x00000005

- *3 unknown/js code*
- [`LLIntSlowPaths.cpp:968`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/llint/LLIntSlowPaths.cpp#L968>): `llint_slow_path_get_by_id`
- *2 unknown/js code*
- `llint_entry`

**Comment by dapabe** [comment]
Updated log output on v1.1.24:
```
Bun v1.1.24 (85a32991) Windows x64
Windows v.win10_fe
CPU: sse42 avx avx2
Args: "C:\Program Files\nodejs\\node_modules\bun\bin\bun.exe" "--env-file=.env.development" "--sourcemap" "dist/index.js" "--skip-server"
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:constants" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:module" "node:net" "node:os" "node:path" "node:process" "node:readline" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "utf-8-validate" "node:http2" "node:dgram"
Elapsed: 5741ms | User: 656ms | Sys: 93ms
RSS: 0.20GB | Peak: 0.22GB | Commit: 0.24GB | Faults: 62232

panic(main thread): Segmentation fault at address 0x5
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.24/wa185a3299A6ogggD_____wzsvjB___A2AK

```

**Comment by Jarred-Sumner** [comment]
Not sure yet how to reproduce it, as I think it's missing credentials

**Comment by heimskr** [comment]
Possibly fixed by #15357.
